### PR TITLE
feat: Latest audited EP0.7 BW MSCA contract scripts and deployments

### PR DESCRIPTION
**Pull request by ashutosh-ukey**
## Summary

Add transfer ownership script and deployment logs for remaining BW MSCA smart contracts across mainnets and testnets.

### Checklist
- [x] Did you add new tests and confirm all tests pass? (`yarn test`)
- [x] Did you update relevant docs? (docs are found in the `docs` folder)
- [x] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [x] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [x] Did you run lint (`yarn lint`) and fix any issues?
- [x] Did you run formatter (`yarn format:check`) and fix any issues (`yarn format:write`)?

## Testing

Did sanity testing using our BW systems on testnets: wallet creation, accepting NFTs, initiating a transfer out.

### CODE REVIEW

File: broadcast/007_DeployColdStorageAddressBookPlugin.s.sol/1/run-1728075551.json
Comment (ashutosh-ukey) [inline broadcast/007_DeployColdStorageAddressBookPlugin.s.sol/1/run-1728075551.json:1]: I unfortunately accidentally lost the deployment logs for the mainnet `ARB` and `POLYGON` chains while making git changes, but the Cold Stroage Address Book Plugin contract has still been deployed / verified on those chains, and the deployment tx hash can be found on their corresponding block scan links:
- https://polygonscan.com/address/0x3c95978Af08B6B2Fd82659B393be86AfB4bd3D6F
- https://arbiscan.io/address/0x3c95978Af08B6B2Fd82659B393be86AfB4bd3D6F

File: script/010_StakeUpgradableMSCAFactory.s.sol [hunk 1]
```diff
@@ -30,7 +30,9 @@ contract StakeUpgradableMSCAFactory is Script {
         uint256 key = vm.envUint("MSCA_FACTORY_OWNER_PRIVATE_KEY");
 
         // Configure stake (using minimums from https://docs.alchemy.com/docs/bundler-services#minimum-stake)
-        uint256 stakeValue = 10 ether; // For MATIC Amoy (make sure to verify this value if you switch chains)
+        // uint256 stakeValue = 10 ether; // For MATIC Amoy (make sure to verify this value if you switch chains)
```
Comment (huaweigu) [inline script/010_StakeUpgradableMSCAFactory.s.sol:33]: minor `MATIC Amoy` -> `Polygon Amoy`

File: script/010_StakeUpgradableMSCAFactory.s.sol [hunk 2]
```diff
@@ -30,7 +30,9 @@ contract StakeUpgradableMSCAFactory is Script {
...
+        // uint256 stakeValue = 100 ether; // For MATIC
```
Comment (huaweigu) [inline script/010_StakeUpgradableMSCAFactory.s.sol:34]: ditto on Polygon

File: script/011.1_SetUpgradableMSCAFactoryPlugins.s.sol [hunk 3]
```diff
@@ -39,14 +38,14 @@ contract SetUpgradableMSCAFactoryPlugins is Script {
...
         bool[] memory pluginPermissions = new bool[](numPlugins);
 
-        plugins[0] = SINGLE_OWNER_PLUGIN_ADDRESS;
-        plugins[1] = COLD_STORAGE_ADDRESS_BOOK_PLUGIN_ADDRESS;
-        plugins[2] = WEIGHTED_MULTISIG_PLUGIN_ADDRESS;
-        plugins[3] = DEFAULT_TOKEN_CALLBACK_PLUGIN_ADDRESS;
+        plugins[0] = COLD_STORAGE_ADDRESS_BOOK_PLUGIN_ADDRESS;
+        plugins[1] = WEIGHTED_MULTISIG_PLUGIN_ADDRESS;
+        plugins[2] = DEFAULT_TOKEN_CALLBACK_PLUGIN_ADDRESS;
+        // plugins[3] = SINGLE_OWNER_PLUGIN_ADDRESS;
```
Comment (huaweigu) [inline script/011.1_SetUpgradableMSCAFactoryPlugins.s.sol:48]: nitpick: removing this commented out line `// plugins[3]`? we can add it if needed in the future

File: script/013_TransferUpgradableMSCAFactoryOwnership.s.sol [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+pragma solidity 0.8.24;
+
+import {UpgradableMSCAFactory} from "../src/msca/6900/v0.7/factories/UpgradableMSCAFactory.sol";
+
+import {UPGRADABLE_MSCA_FACTORY_ADDRESS} from "./000_ContractAddress.sol";
+import {Script, console} from "forge-std/src/Script.sol";
+
+contract StakeUpgradableMSCAFactory is Script {
```
Comment (huaweigu) [inline script/013_TransferUpgradableMSCAFactoryOwnership.s.sol:26]: could we please rename the contract name to `TransferUpgradableMSCAFactoryOwnership`?
also could this be 011.2 (unlike flyway migration)

File: script/verify/UpgradableMSCA/abiEncodedConstructorArgs [hunk 5]
```diff
@@ -0,0 +1 @@
+0000000000000000000000000000000071727de22e5e9d8baf0edac6f37da03200000000000000000000000001d716c27bfbee5199511b7f48aa2e01ccd36a10
```
Comment (huaweigu) [inline script/verify/UpgradableMSCA/abiEncodedConstructorArgs:1]: style: missing new line

File: script/verify/UpgradableMSCAFactory/abiEncodedConstructorArgs [hunk 6]
```diff
@@ -0,0 +1 @@
+0000000000000000000000000166ea90e565476f13c6a0d25ed2c35599e587850000000000000000000000000000000071727de22e5e9d8baf0edac6f37da03200000000000000000000000001d716c27bfbee5199511b7f48aa2e01ccd36a10
```
Comment (huaweigu) [inline script/verify/UpgradableMSCAFactory/abiEncodedConstructorArgs:1]: missing new line

**Review by huaweigu — APPROVED** [review]

**Pull request merged**